File: plugins/e200z4/core/p2acFileReader.cpp
#include "arch.h"
#include "codeReader.h"
#include <stdlib.h>

using namespace std;

class codeReader;

int
main (int argc, char *argv[])
{
  if (argc != 2)
    {
      cerr << "Usage ./p2aFileReader code_file" << endl;
      exit (EXIT_FAILURE);
    }

  arch a;
  a.readCodeFile (argv[1]);
  u32 address = a.programCounter ();
  cout << "start " << address << endl;
  
  codeReader *reader = a.getCodeReader ();
  unsigned int const nbCodeSection = reader -> getNbCodeSection ();
  for (int i = 0; i < nbCodeSection; ++i)
    {
      codeSection *section = reader -> getCodeSection (i);
      address = section -> v_addr ();
      u32 addressEnd = address + section -> size ();
      while (address < addressEnd)
	{
	  string info = a.getInstructionStaticInfo (address);
	  cout << info << endl;
	}
    }
  
  return 0;
}

File: plugins/e200z4/core/p2acFileReader.py
#!/usr/bin/python
# vim: set fileencoding=utf-8 :
#
# Read an application file (.elf) and:
#  * decode binary part
#  * export static instruction information: register read/write, branch inst and so on.

# usage ./p2aFileReader <code_file>

import sys
import os
from e200z4 import arch

if len(sys.argv) != 2:
	print "Usage ./p2aFileReader code_file"
	os._exit(1)

f=arch()
f.readCodeFile(sys.argv[1])
print "start %d" % f.programCounter()
address=f.programCounter()
for i in range(f.getCodeReader().getNbCodeSection()):
	section = f.getCodeReader().getCodeSection(i);
	address = section.v_addr();
	addressEnd = address + section.size();
	while(address < addressEnd):
		info,address=f.getInstructionStaticInfo(address)
		print info

